components/BottomSheet/src/MDCBottomSheetPresentationController.m

```diff
@@ -179,9 +179,11 @@ - (void)viewWillTransitionToSize:(CGSize)size
 
  @param preferredSheetHeight If positive, the new value for @sheetView.preferredSheetHeight.
  */
-- (void)setPreferredSheetHeight:(CGFloat)preferredSheetHeight {
+- (void)updatePreferredSheetHeight:(CGFloat)preferredSheetHeight {
```

Comment:
I believe #5125 updated this method to make it more testable, I can revert that change if you'd like.

Reply:
Ah! I see.

I think we don't need to revert that change - you're increasing the surface area of the APIs here to include both a setter and a "did change" method that commits the results to the sheetView.

Prior to this PR, the tests relied on the method previously named `setPreferredSheetHeight`. The tests were using this method to propagate changes to the sheetView.

After this PR, `setPreferredSheetHeight` has a different meaning and you're introducing a new method, `updatePreferredSheetHeight` that models the old meaning of `setPreferredSheetHeight`.

You can still test as before, but your existing tests will now call the `updatePreferredSheetHeight` API to propagate changes to the sheetView instead of `setPreferredSheetHeight`, except in cases where you're testing the new property's behavior specifically.

Reply:
Tying this back to my original suggestion, if you remove `updatePreferredSheetHeight`'s argument then your tests will still be able to call this method, but in doing so they must set the relevant properties that they are testing.

Reply:
Spoke over gvc:

1. Partial revert of #5125 so that we still have an `updatePreferredSheetHeight` method that doesn't accept arguments.
2. `updatePreferredSheetHeight` will be exposed to tests so that we can "flush" the calculated sheet height to sheetView once we've configured either `preferredSheetHeight` or `preferredContentSize.height`.

Reply:
Also discussed the possibility of making `updatePreferredSheetHeight` more testable by making it stateless, but that doing so can be a follow-up exploration:

```objc
+ (CGFloat)preferredSheetHeightForViewController:(UIViewController *)viewController
                                           frame:(CGRect)frame
                        withPreferredSheetHeight:(CGFloat)preferredSheetHeight {
  CGFloat calculatedPreferredSheetHeight = viewController.preferredContentSize.height;
  if (preferredSheetHeight > 0.f) {
    calculatedPreferredSheetHeight = preferredSheetHeight;
  }
  if (MDCCGFloatEqual(preferredSheetHeight, 0)) {
    preferredSheetHeight = MDCRound(CGRectGetHeight(frame) / 2);
  }
  self.sheetView.preferredSheetHeight = preferredSheetHeight;
}
```